simple/stats/events_importer.py

```diff
@@ -259,3 +301,14 @@ def _write_debug_csvs(self) -> None:
                    self.debug_resolve_fh)
       self.debug_resolve_fh.write_string(
           self.debug_resolve_df.to_csv(index=False))
+
+
+# Utility methods
+def _time_period(date_str: str, period: TimePeriod) -> str:
```

Comment:
Since this is applied on each cell and there could be a lot of duplicate (if all from one year). Not sure if parsing date on each cell would be a performance issue. But maybe this can be optimized by having a "cache" object and a return an inner nested function like:

def _time_period():
  cache = {}
  def actual_func(...):
     # use and check cache.
      ...
   return actual_func

Reply:
github removes indention. `actual_func()` should be an inner function of _time_period and `cache` is captured.